### DISCUSSION TITLE
What are the advantages of flashinfer.sampling over the torch version?

### DISCUSSION THREAD

**Discussion by CSEEduanyu** [sgl-project/sglang#1266] [General]
What are the advantages of flashinfer.sampling over the torch version?

**Comment by yzh119** [comment]
Flashinfer implement the sampling algorithms in a way that do not require sorting, which avoids writing to global memory.

And the fused kernel reduces kernel launching overhead (the original torch implementation launches dozens of cuda kernels), which is not efficient.
